### PR TITLE
ARK_GPS: porting to ArduPilot

### PR DESCRIPTION
**Pull request by samchamberlin**
Added support of the ARK_GPS board on ArduPilot. All sensors are running.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_GPS/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+define HAL_PERIPH_ENABLE_MAG
+define HAL_PERIPH_ENABLE_BARO
+define HAL_PERIPH_ENABLE_BUZZER
+define HAL_PERIPH_ENABLE_RC_OUT
+define HAL_PERIPH_ENABLE_NOTIFY
+
+# GPS on 2nd port
+define HAL_PERIPH_GPS_PORT_DEFAULT 1
+
+# TODO: params changed in CUAV_GPS we might want
```
Comment (khancyr) [inline libraries/AP_HAL_ChibiOS/hwdef/ARK_GPS/hwdef.dat:125]: Do you still need to discuss those ?

**Comment by hendjoshsr71** [comment]
Hi Sam!  

Would you please change the commit messages to something like, "hwdef: add ARK_GPS"

We usually prepend each commit with the library each file(s) changed is from, but for hwdef's "hwdef:" is the practice.

**Review by khancyr — COMMENTED** [review]
Welcome,
I was looking forward to have support for those board and that is even better when the manufacturer is able to do the support itself.

I am not qualified enough for the content, but we got a strict rule on commit message.

Could you change your commit message to start with AP_HAL_Chibios: ?
If you need some assistance on git, there are some infos on the wiki and you would find someone to help on discord.

Thanks

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_GPS/hwdef.dat (hunk 1 above)
Comment (samchamberlin) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_GPS/hwdef.dat]: Nope, removed. Thanks!

**Comment by samchamberlin** [comment]
Thank you @hendjoshsr71 & @khancyr !

**Review by tridge — APPROVED** [review]

**Pull request merged**